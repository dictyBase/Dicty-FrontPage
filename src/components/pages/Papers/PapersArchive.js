// @flow
import React from "react"
import { Flex, Box } from "rebass"
import FontAwesome from "react-fontawesome"
import { Banner, Header, Hdrtxt } from "styles/EditablePageStyles"
import { Link } from "styles"
import papers from "data/papers"

/**
 * This displays the Dicty papers archive.
 */

const PapersArchive = () => {
  const paperList = papers.map(paper => (
    <Flex justify="center" key={paper.title}>
      <Box w={["90%", "90%", "90%", "60%"]} py={5}>
        <div>{paper.author}</div>
        <div>
          <strong>{paper.title}</strong>
        </div>
        <div>
          <em>{paper.journal}</em>
          <Link href={paper.link} target="new">
            {" "}
            Pubmed
          </Link>
        </div>
      </Box>
    </Flex>
  ))
  return (
    <div>
      <Banner style={{ backgroundColor: "#E0F2F7" }}>
        <Header style={{ backgroundColor: "#E0F2F7" }}>
          <FontAwesome name="paperclip" /> Dicty Papers
        </Header>
        <Hdrtxt>
          Papers on <em>Dictyostelium</em> in the last 5 years
        </Hdrtxt>
      </Banner>
      <br />
      <Flex justify="center">
        <Box>{paperList}</Box>
      </Flex>
    </div>
  )
}

export default PapersArchive
