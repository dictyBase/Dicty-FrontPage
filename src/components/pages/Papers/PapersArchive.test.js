import React from "react"
import toJson from "enzyme-to-json"
import { shallow } from "enzyme"
import "jest-styled-components"
import PapersArchive from "./PapersArchive"

test("matching a snapshot of PapersArchive", () => {
  const subject = shallow(<PapersArchive />)
  expect(toJson(subject)).toMatchSnapshot()
})
