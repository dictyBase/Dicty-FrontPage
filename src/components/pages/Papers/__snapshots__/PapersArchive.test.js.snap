// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of PapersArchive 1`] = `
<div>
  <styled.div
    style={
      Object {
        "backgroundColor": "#E0F2F7",
      }
    }
  >
    <styled.h1
      style={
        Object {
          "backgroundColor": "#E0F2F7",
        }
      }
    >
      <FontAwesome
        name="paperclip"
      />
       Dicty Papers
    </styled.h1>
    <styled.p>
      Papers on 
      <em>
        Dictyostelium
      </em>
       in the last 5 years
    </styled.p>
  </styled.div>
  <br />
  <Flex
    justify="center"
  >
    <Box>
      <Flex
        justify="center"
        key="Overexpression and purification of human myosins from transiently and stably transfected suspension adapted HEK293SF-3F6 cells."
      >
        <Box
          py={5}
          w={
            Array [
              "90%",
              "90%",
              "90%",
              "60%",
            ]
          }
        >
          <div>
            Usaj, Zattelman, Regev, Shneyer, Wiesel-Motiuk & Henn
          </div>
          <div>
            <strong>
              Overexpression and purification of human myosins from transiently and stably transfected suspension adapted HEK293SF-3F6 cells.
            </strong>
          </div>
          <div>
            <em>
              Anal Biochem. 2018 Jul 31. pii: S0003-2697(18)30419-6. doi: 10.1016/j.ab.2018.07.026. [Epub ahead of print]
            </em>
            <Link
              href="https://www.ncbi.nlm.nih.gov/pubmed/30075102"
              target="new"
            >
               
              Pubmed
            </Link>
          </div>
        </Box>
      </Flex>
      <Flex
        justify="center"
        key="Assaying Rho GTPase-Dependent Processes in Dictyostelium discoideum."
      >
        <Box
          py={5}
          w={
            Array [
              "90%",
              "90%",
              "90%",
              "60%",
            ]
          }
        >
          <div>
            Marinović M, Xiong H, Rivero F, Weber I.
          </div>
          <div>
            <strong>
              Assaying Rho GTPase-Dependent Processes in Dictyostelium discoideum.
            </strong>
          </div>
          <div>
            <em>
              Methods Mol Biol. 2018;1821:371-392. doi: 10.1007/978-1-4939-8612-5_25.
            </em>
            <Link
              href="https://www.ncbi.nlm.nih.gov/pubmed/30062425"
              target="new"
            >
               
              Pubmed
            </Link>
          </div>
        </Box>
      </Flex>
      <Flex
        justify="center"
        key="Promoter-mediated diversification of transcriptional bursting dynamics following gene duplication."
      >
        <Box
          py={5}
          w={
            Array [
              "90%",
              "90%",
              "90%",
              "60%",
            ]
          }
        >
          <div>
            Tunnacliffe E, Corrigan AM, Chubb JR.
          </div>
          <div>
            <strong>
              Promoter-mediated diversification of transcriptional bursting dynamics following gene duplication.
            </strong>
          </div>
          <div>
            <em>
              Proc Natl Acad Sci U S A. 2018 Jul 30. pii: 201800943. doi: 10.1073/pnas.1800943115. [Epub ahead of print]
            </em>
            <Link
              href="https://www.ncbi.nlm.nih.gov/pubmed/30061408"
              target="new"
            >
               
              Pubmed
            </Link>
          </div>
        </Box>
      </Flex>
      <Flex
        justify="center"
        key="Expression of N471D strumpellin leads to defects in the endolysosomal system."
      >
        <Box
          py={5}
          w={
            Array [
              "90%",
              "90%",
              "90%",
              "60%",
            ]
          }
        >
          <div>
            Song L, Rijal R, Karow M, Stumpf M, Hahn O, Park L, Insall R, Schröder R, Hofmann A, Clemen CS, Eichinger L.
          </div>
          <div>
            <strong>
              Expression of N471D strumpellin leads to defects in the endolysosomal system.
            </strong>
          </div>
          <div>
            <em>
              Dis Model Mech. 2018 Jul 30. pii: dmm.033449. doi: 10.1242/dmm.033449. [Epub ahead of print]
            </em>
            <Link
              href="https://www.ncbi.nlm.nih.gov/pubmed/30061306"
              target="new"
            >
               
              Pubmed
            </Link>
          </div>
        </Box>
      </Flex>
      <Flex
        justify="center"
        key="Adaptation of Living Systems."
      >
        <Box
          py={5}
          w={
            Array [
              "90%",
              "90%",
              "90%",
              "60%",
            ]
          }
        >
          <div>
            Tu Y, Rappel WJ.
          </div>
          <div>
            <strong>
              Adaptation of Living Systems.
            </strong>
          </div>
          <div>
            <em>
              Annu Rev Condens Matter Phys. 2018 Mar;9:183-205. doi: 10.1146/annurev-conmatphys-033117-054046. Epub 2017 Dec 8.
            </em>
            <Link
              href="https://www.ncbi.nlm.nih.gov/pubmed/30057689"
              target="new"
            >
               
              Pubmed
            </Link>
          </div>
        </Box>
      </Flex>
    </Box>
  </Flex>
</div>
`;
